### PR TITLE
Added MUQ build recipe.

### PR DESCRIPTION
**Pull request by mparno**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mparno** [comment]
@conda-forge/staged-recipes  This recipe is ready for review whenever you have a chance.  Thanks!

File: recipes/muq/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+  host:
+    - python
+    - numpy
+    - hdf5
+    - boost-cpp ==1.74
```
Comment (chrisburr) [inline recipes/muq/meta.yaml:24]: This should be allowed to come from the global conda-forge pinning:
```suggestion
    - boost-cpp
```

File: recipes/muq/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - eigen >=3.3
+    - sundials ==5.4.0
+    - nanoflann ==1.3.1
+    - nlopt
+  run:
+    - python
+    - numpy
```
Comment (chrisburr) [inline recipes/muq/meta.yaml:31]: https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/muq/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - hdf5
+    - boost-cpp ==1.74
```
Comment (chrisburr) [inline recipes/muq/meta.yaml:33]:
```suggestion
    - boost-cpp
```

**Comment by mparno** [comment]
Thanks @chrisburr for your comments.  I believe I've addressed them all and this pull request is now ready for another review.

**Pull request merged**